subsys/bluetooth/controller/hci/hci.c

```diff
@@ -211,9 +211,9 @@ static uint32_t cis_pending_count;
 static uint64_t event_mask = DEFAULT_EVENT_MASK;
 static uint64_t event_mask_page_2 = DEFAULT_EVENT_MASK_PAGE_2;
 static uint64_t le_event_mask = DEFAULT_LE_EVENT_MASK;
-#if defined(CONFIG_BT_HCI_VS_EVT)
-static uint64_t vs_events_mask = DEFAULT_VS_EVT_MASK;
-#endif /* CONFIG_BT_HCI_VS_EVT */
+#if defined(CONFIG_BT_HCI_VS)
+__maybe_unused static uint64_t vs_events_mask = DEFAULT_VS_EVT_MASK;
+#endif /* CONFIG_BT_HCI_VS */
```

Comment:
Is there any chance we can remove the guards altogether? 

IIRC if a `static` variable is only by `static` functions and the calls to the functions are guarded by `IS_ENABLED`, then the linker should be able to remove the `static` variables as well. 

Generally `#if defined` is only really necessary when fields inside structs are used. Is this the case here?

Reply:
> Is there any chance we can remove the guards altogether?
> 
> IIRC if a `static` variable is only by `static` functions and the calls to the functions are guarded by `IS_ENABLED`, then the linker should be able to remove the `static` variables as well.
> 
> Generally `#if defined` is only really necessary when fields inside structs are used. Is this the case here?

You're mostly right. In this case the necessity comes from direction finding vendor events, whose implementation is guarded by #ifdefs and calls to which come from `case` statements which are also guarded by #ifdefs. So I decided to leave these out of the scope of this PR.

Reply:
Btw, one more push coming once CI shows green, since I decided to mention these in the migration guide as well (at least some of the changes can be seen as API breakage, or at least something that can cause build breakage in downstream trees).